src/sseq.rs

```diff
@@ -96,9 +96,56 @@ impl<const N: usize> SSeqGen<N> {
         res
     }
 
+    /// Iterator over sequences one substitution (hamming distance) away.
     pub fn one_hamming_iter(self, opt: HammingIterOpt) -> SSeqOneHammingIter<N> {
         SSeqOneHammingIter::new(self, opt)
     }
+
+    /// Iterator over sequences one deletion away.
+    pub fn one_deletion_iter(self) -> impl Iterator<Item = Self> {
+        (0..self.len()).map(move |i| {
+            let mut seq = self;
+            seq.remove(i);
+            seq
+        })
+    }
```

Comment:
I think you can remove one memcopy here by accepting `&self` and dereferencing in your copy assignment.  Probably unimportant for ultimate performance, and if N <=8 its better to keep your existing implementation.  Mostly just thinking out loud here.  If N is commonly >= 16, it might be better to use the borrowed receiver version.  I see above that we newtype a common case where N = 23.
```suggestion
    pub fn one_deletion_iter(&self) -> impl Iterator<Item = Self> {
        (0..self.len()).map(move |i| {
            let mut seq = *self;
            seq.remove(i);
            seq
        })
    }
```

Reply:
This is true for all of the by-value receivers in this package, since accepting `self` implies copying the entire struct into the function arguments, but accepting `&self` only requires writing a pointer.

Of course, optimizations will do any number of things that could invalidate these surface-level observations.  I'm not sure if there's any special-case handling around immutable references to `Copy` types.

Reply:
A lot of these functions may end up being inlined and so the distinction may not even matter.

Reply:
Good point -- I have been on the fence on whether to accept by copy or reference given that the most common value of N is ~16. Might be better to change to `&self`